js/pro/wazirx.js

```diff
@@ -0,0 +1,607 @@
+            const available = this.safeNumber (balance, 'b');
+            const locked = this.safeNumber (balance, 'l');
+            const account = this.account ();
+            account['free'] = available;
+            account['used'] = locked;
+            this.balance[code] = account;
+        }
+        this.balance = this.safeBalance (this.balance);
+        const messageHash = 'balance';
+        client.resolve (this.balance, messageHash);
+    }
+
+    parseWSTrade (trade, market = undefined) {
+        //
+        //     {
+        //         "E": 1631683058000,
+        //         "S": "ask",
+        //         "U": "inr",
+        //         "a": 114144050,
+        //         "b": 114144121,
+        //         "f": "0.2",
+        //         "m": true,
+        //         "o": 26946170,
+        //         "p": "5.0",
+        //         "q": "20.0",
+        //         "s": "btcinr",
+        //         "t": 17376032,
+        //         "w": "100.0"
+        //     }
+        //
+        const timestamp = this.safeNumber (trade, 'E');
+        const marketId = this.safeString (trade, 's');
+        market = this.safeMarket (marketId, market);
+        const feeCurrencyId = this.safeString (trade, 'U');
+        return this.safeTrade ({
+            'id': this.safeString (trade, 'id'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market['symbol'],
+            'order': this.safeString (trade, 'o'),
+            'type': undefined,
+            'side': undefined,
+            'takerOrMaker': this.safeValue (trade, 'm'),
+            'price': this.safeString (trade, 'p'),
+            'amount': this.safeString (trade, 'q'),
+            'cost': undefined,
+            'fee': {
+                'cost': this.safeString (trade, 'f'),
+                'currency': this.safeCurrencyCode (feeCurrencyId),
+                'rate': undefined,
+            },
+        }, market);
+    }
+
+    async watchTicker (symbol, params = {}) {
+        /**
+         * @method
+         * @name wazirx#watchTicker
+         * @description watches a price ticker, a statistical calculation with the information calculated over the past 24 hours for a specific market
+         * @see https://docs.wazirx.com/#all-market-tickers-stream
+         * @param {string} symbol unified symbol of the market to fetch the ticker for
+         * @param {object} params extra parameters specific to the wazirx api endpoint
+         * @returns {object} a [ticker structure]{@link https://docs.ccxt.com/en/latest/manual.html#ticker-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const url = this.urls['api']['ws'];
+        const messageHash = 'ticker:' + market['symbol'];
+        const subscribeHash = 'tickers';
+        const subscribe = {
+            'event': 'subscribe',
+            'streams': [ '!ticker@arr' ],
```

Comment:
our regex will blindly convert `!something` to `not something` so this will be converted to `not ticker` 😅 
In these cases we should make it with a string concatenation: `'!' + 'ticker'`